packages/grafana-ui/src/components/Segment/Segment.tsx

```diff
@@ -18,18 +18,23 @@ export function Segment<T>({
   const [Label, width, expanded, setExpanded] = useExpandableLabel(false);
 
   if (!expanded) {
-    return <Label Component={Component || <a className={cx('gf-form-label', 'query-part', className)}>{value}</a>} />;
+    return (
+      <Label
+        Component={Component || <a className={cx('gf-form-label', 'query-part', className)}>{value && value.label}</a>}
```

Comment:
maybe label could have been looked up from the options

Reply:
Wouldn't work for SegmentAsync since it lazyloads options